heron/spi/src/java/com/twitter/heron/spi/utils/LauncherUtils.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.spi.utils;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.twitter.heron.api.generated.TopologyAPI;
+import com.twitter.heron.common.basics.SysUtils;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Keys;
+import com.twitter.heron.spi.packing.IPacking;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.scheduler.IScheduler;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+
+/**
+ * {@link LauncherUtils} contains helper methods used by the server and client side launch
+ * controllers; {@link com.twitter.heron.scheduler.LaunchRunner} and
+ * {@link com.twitter.heron.scheduler.SchedulerMain}
+ */
+public class LauncherUtils {
+  private static final Logger LOG = Logger.getLogger(LauncherUtils.class.getName());
+
+  /**
+   * An instance of this utility class to avoid the need of injecting a factory when mocking utils.
+   */
+  @VisibleForTesting
+  public static LauncherUtils instance = new LauncherUtils();
+
+  /**
+   * For testing only. Could be used to setup a mock
+   */
+  @VisibleForTesting
+  public static void setInstance(LauncherUtils instance) {
```

Comment:
Ok. Will fix it. In some cases the utils user object is not visible. So it may not be feasible to set the mock in those objects. As of now the launchers do not have this issue.